src/apps/bm_devkit/serial_payload_example/user_code/user_code.cpp

```diff
@@ -33,9 +36,53 @@ static u_int32_t uart_mode_config = DEFAULT_UART_MODE;
 // A buffer for our data from the payload uart
 char payload_buffer[2048];
 
+void sdi_wake(void) {
```

Comment:
@towynlin @victorsowa12 @matt001k - `sdi_wake` and `sdi_break_mark` are functions that are required for managing serial transactions with an SDI-12 sensor.

@uma-kat and I think these should probably go into the Bristlefin namespace as these are dealing with specific pins and the specific behavior of the SDI-12 transceiver on the BF board, but that would bring PLUART into bristlefin.h.

Please let us know if you think these functions should go somewhere else - PLUART directly? new file that includes BF and SDI12?...

Reply:
We could also abstract out some of the logic in setup like `bristlefin::setup_rs485_hd()`, `bristlefin::setup_sdi12()`, which would be nice.

Reply:
I don't have any issues with it. Makes sense to put the SDI-12 stuff into the bristlefin namespace since bristlefin's are the only board that can use SDI-12. Also no problem for pulling in PLUART since it is the shared library for all board types.

Reply:
I feel like these functions themselves should be located outside of this file. Analyzing the bristlefin class,  I feel like this would be the perfect location currently for the API  mentioned as this is what houses `sdi12Tx/sdi12Rx`. Personally I do not see an issue bringing in PLUART, and I doubt that it would have to be included in the header file.

Personally I believe it would be nice to see all of the code encapsulated in a single module, is there any reason why we could not use the `exo3s_sensor.h` API here? Maybe that file can be moved to somewhere in `src/lib/drivers` and then be used in this file directly?
Or is `sdi_wake` and `sdi_break_mark` generic for every type of SDI12 type device?
If that is the case, could we create a module called `sdi12.h` in `src/lib/drivers` and utilize that here?
Maybe as a future task configure that driver to be agnostic to the platform used such as the pins, etc.?

Reply:
Thanks @victorsowa12 and @matt001k, `sdi_wake` and `sdi_break_mark` are generic for all SDI12 devices, and the implementation in these functions is specific to Mote + Bristlefin - not specific to the EXO3, and not abstract enough to work on a different platform + interface board.

As a first step I like migrating these to bristlefin.h and updating exo3_sensor.h to use those functions.

I'll give that a go and poke for re-review when it's ready for a look.